test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -463,4 +514,156 @@ static public void testGetSizeOfZeroSizeVT() {
 			assertEquals(size, 8);
 		}
 	}
+
+	@Test(dataProvider = "compareAndDoSomethingFuncs")
+	static public void testCompareAndSetNullObj(CompareAndDoSomethingFunction compareAndSwapValue) throws Throwable {
+		if (isFlatteningEnabled) {
+			assertThrows(NullPointerException.class, () -> {
+				compareAndSwapValue.execute(null, vtPointOffsetX, ValueTypeInt.class, vtPoint.x, new ValueTypeInt(1));
+			});
+		}
+	}
+
+	@Test(dataProvider = "compareAndDoSomethingFuncs")
+	static public void testCompareAndSetNullClz(CompareAndDoSomethingFunction compareAndSwapValue) throws Throwable {
+		if (!isFlatteningEnabled) {
+			compareAndSwapValue.execute(vtPoint, vtPointOffsetX, null, vtPoint.x, new ValueTypeInt(1));
+			assertEquals(vtPoint.x.i, 1);
+		}
+	}
+
+	@Test(dataProvider = "compareAndDoSomethingFuncs")
+	static public void testCompareAndSetNullV1(CompareAndDoSomethingFunction compareAndSwapValue) throws Throwable {
+		int original = vtPoint.x.i;
+		boolean success = compareAndSwapValue.execute(vtPoint, vtPointOffsetX, ValueTypeInt.class, null, new ValueTypeInt(1));
+		assertFalse(success);
+		assertEquals(vtPoint.x.i, original);
+	}
+
+	@Test(dataProvider = "compareAndDoSomethingFuncs")
+	static public void testCompareAndSetNullV2(CompareAndDoSomethingFunction compareAndSwapValue) throws Throwable {
+		int original = vtPoint.x.i;
+		boolean success = compareAndSwapValue.execute(vtPoint, vtPointOffsetX, ValueTypeInt.class, new ValueTypeInt(original + 1), null);
+		assertFalse(success);
+		assertEquals(vtPoint.x.i, original);
+
+		if (isFlatteningEnabled) {
+			assertThrows(NullPointerException.class, () -> {
+				compareAndSwapValue.execute(vtPoint, vtPointOffsetX, ValueTypeInt.class, vtPoint.x, null);
+			});
+		} else {
+			success = compareAndSwapValue.execute(vtPoint, vtPointOffsetX, ValueTypeInt.class, vtPoint.x, null);
+			assertTrue(success);
+			assertEquals(vtPoint.x, null);
+		}
+	}
+
+	@Test(dataProvider = "compareAndDoSomethingFuncs")
+	static public void testCompareAndSetObjectXSuccess(CompareAndDoSomethingFunction compareAndSwapValue) throws Throwable {
```

Comment:
The name of the function has "Object" in it. But the passed in parameter is "compareAndSwapValue". Seems confusing.

Reply:
Okay, I can rename it to `testCompareAndSetPointXSuccess` to make it more clear

Reply:
updated in 8add8f2